File: apps/api/src/domain/value_objects/verdict.py
"""성분/제품 판정 등급."""

from __future__ import annotations

from enum import StrEnum


class Verdict(StrEnum):
    """개별 성분 또는 전체 결과에 대한 판정 등급.

    단정형이 아닌 참고 판단형 등급이다.
    """

    GOOD = "GOOD"
    CAUTION = "CAUTION"
    BAD = "BAD"

File: apps/mobile/src/entities/analysis/model/verdict.ts
import type { AnalysisResult } from '@cosmetics-analyzer/api-client';

export type Verdict = AnalysisResult['verdict'];
export type Recommendation = AnalysisResult['recommendation'];

/** Verdict 색상 토큰 키 (ui-tokens 의 verdict.* 와 매핑). */
export type VerdictTone = 'good' | 'caution' | 'bad';

// 참고 판단형 한글 라벨. 단정 표현("유해"/"위험"/"안전")은 사용하지 않는다 (PDD §13).
export const VERDICT_LABEL: Record<Verdict, string> = {
  GOOD: '양호',
  CAUTION: '주의',
  BAD: '유의',
};

export const RECOMMENDATION_LABEL: Record<Recommendation, string> = {
  RECOMMENDED: '참고로 추천',
  NEUTRAL: '중립',
  CAUTION_NEEDED: '주의해서 확인',
};

const VERDICT_TONE: Record<Verdict, VerdictTone> = {
  GOOD: 'good',
  CAUTION: 'caution',
  BAD: 'bad',
};

export function verdictTone(verdict: Verdict): VerdictTone {
  return VERDICT_TONE[verdict];
}
